### ISSUE TITLE
Partitions

### ISSUE THREAD

**Issue by adrianmihalko** [dustindclark/homebridge-envisalink#37]
Hi,

I have a 2 partition system, is this supported? Can I arm individually the two partitions? I think it should be shown as two separate alarm in the Homekit.

There is a "partitions" part in the config file, but I don't know how to define multiple partitions.

```
    {
      "platform": "Envisalink",
      "host": "192.168.1.54",
      "deviceType": "DSC",
      "password": "user",
      "pin": "1955",
      "suppressZoneAccessories": false,
      "suppressClockReset": true,
      "partitions": [
        {
          "name": "Alarm"
        }
      ],
      "zones": [
        {
          "name": "Home/door",
          "type": "door",
          "partition": 2,
          "zoneNumber": 10
        },
        {
          "name": "Home/motion",
          "type": "motion",
          "partition": 2,
          "zoneNumber": 9
        },
        {
          "name": "Shop - front door",
          "type": "door",
          "partition": 1,
          "zoneNumber": 1
        },
```

**Comment by adrianmihalko** [comment]
Ok, I just figured out that it is easy as:

```
"partitions": [
        {
          "name": "Partition 1"
        },
        {
          "name": "Partition 2"
        }
      ],

```

Everything is working great, except that Alarm status is not updated.

```
Mar 24 13:09:35 ubuntu homebridge[29653]: Partition 2 Ready
Mar 24 13:09:35 ubuntu homebridge[29653]: Setting Obstructed true
Mar 24 13:09:35 ubuntu homebridge[29653]: [3/24/2018, 1:09:35 PM] [Envisalink] Attempting to set alarm state to:  0
Mar 24 13:09:35 ubuntu homebridge[29653]: [3/24/2018, 1:09:35 PM] [Envisalink] Arming alarm to Stay/Night.
Mar 24 13:09:35 ubuntu homebridge[29653]: sendcommand 0312C6
Mar 24 13:09:35 ubuntu homebridge[29653]: Command 031 Acknowledged
Mar 24 13:09:36 ubuntu homebridge[29653]: Exit Delay of Partition 2 in Progress
Mar 24 13:09:56 ubuntu homebridge[29653]: Partition 2 has been armed by Quick Arm, Auto Arm, Keyswitch, DLS or Wireless Key
Mar 24 13:09:56 ubuntu homebridge[29653]: Partition 21 is Armed
Mar 24 13:10:33 ubuntu homebridge[29653]: Zone 009 is open
Mar 24 13:10:33 ubuntu homebridge[29653]: Set status on accessory Lakás/mozgás to {"name":"Zone Open","pre":"Zone","bytes":3,"post":"is open","se
Mar 24 13:10:37 ubuntu homebridge[29653]: Zone 009 is restored
Mar 24 13:10:37 ubuntu homebridge[29653]: Set status on accessory Lakás/mozgás to {"name":"Zone Alarm Restore","pre":"Zone","bytes":4,"post":"is 

```

I set Home mode alarm, it arms fine in this mode, but Alarm status in Homebridge stays Off. What can be the problem here?

![img_1341](https://user-images.githubusercontent.com/733361/37863854-942a8a06-2f65-11e8-8050-a130acd9f919.png)
![img_1342](https://user-images.githubusercontent.com/733361/37863853-92535bc2-2f65-11e8-852b-6be39c739d4a.png)

**Comment by adrianmihalko** [comment]
If I check raw status messages with telnet I see 652 (Partition Armed) command:

```
madrian@ubuntu:/usr/lib/node_modules/homebridge-envisalink$ telnet localhost 4026
Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.
6562D3
7012CA
6522100
```

Where 652 is the command
2 is the partition
1 is the mode
00 is the checksum.

```
EnvisalinkPlatform.prototype.partitionUpdate = function (data) {
    var watchevents = ['601', '602', '609', '610', '650', '651', '652', '654', '656', '657'];
    if (data.code == "652") {
        //0: AWAY, 1: STAY, 2:  ZERO-ENTRY-AWAY, 3:  ZERO-ENTRY-STAY
        if (data.mode == '1' || data.mode == "3") {
            this.awayStay = Characteristic.SecuritySystemCurrentState.STAY_ARM;
        } else {
            this.awayStay = Characteristic.SecuritySystemCurrentState.AWAY_ARM;
        }
    }

```

If am not wrong, this code set the alarm status in Homebridge, but I have no idea why it's not catching/working 6522100 status message.

**Comment by dustindclark** [comment]
Fixed and published in v0.2.14 thanks to @rafiqsarlie.
